modules/Bio/EnsEMBL/Production/Utils/CopyDatabase.pm

```diff
@@ -631,14 +675,51 @@ sub copy_mysql_files {
   return $copy_failed;
 } 
 
+# Dump the database in /nfs/nobackup/ensembl/ensprod/copy_service
+# Create database on target server
+# Load the database to the target server
+sub copy_mysql_dump {
+  my ($source_dbh,$target_dbh,$source_db,$target_db) = @_;
+  my $copy_failed=0;
+  my $file='/nfs/nobackup/ensembl/ensprod/copy_service/'.$source_db->{host}.'_'.$source_db->{dbname}.'_'.time().'.sql';
+  $logger->info("Dumping $source_db->{dbname} from $source_db->{host} to file $file");
+  if (defined $source_db->{pass}){
+    if ( system("mysqldump --host=$source_db->{host} --user=$source_db->{user} --password=$source_db->{pass} --port=$source_db->{port} $source_db->{dbname} > $file") != 0 ) {
```

Comment:
mysqldump will attempt to lock tables unless you tell it not to; but if you're doing this as the ensro user, you won't have permission to do that. Can use the flag `--skip-lock-tables` to avoid this.

Reply:
Also, I've had trouble in the past with dumping big tables, because the default packet size is too small. That can be changed with something like `--max_allowed_packet=1024M`.

Reply:
@james-monkeyshines: That's two very good points. For the first one, the script already do a table lock here: https://github.com/Ensembl/ensembl-production/blob/fb901c60639e52c27a9b4e424045784f50246717/modules/Bio/EnsEMBL/Production/Utils/CopyDatabase.pm#L194. I think this is quite important and most of the servers now allow ensro user to flush and lock. We actually have steps in our SOP to add these privileges to ensro users using the admin user. So no need to lock twice here.
I forgot about the max_allowed_packed and it did bite me a few times... I've now updated the command. Thanks.